### DISCUSSION TITLE
RabbitMQ looses configuration in docker on a restart

### DISCUSSION THREAD

**Discussion by gabor-lbl** [rabbitmq/rabbitmq-server#8583] [Questions]
### Describe the bug

Running RabbitMQ in docker, with a persistent volume defined for its data will still loose its configuration (queues, etc.) on a restart if `hostname` is not defined for the container. 

If a hostname is not explicitly defined, docker will use a random string. The configuration is saved in `/var/lib/rabbitmq/mnesia/rabbit@$hostname` so it will be different on every restart.

### Reproduction steps

1. Run rabbitmq image without a "hostname" parameter but with a persistent volume mounted at `/var/lib/rabbitmq`
2. Create a queue
3. Restart the container and observe that the queue defined in step 2 is gone

### Expected behavior

Expected behavior is for configuration to survive restarts in docker.

**Comment by lukebakken** [comment]
This is by design. RabbitMQ expects a consistent hostname.

**Reply by gabor-lbl** [reply]
I guess then maybe this should be logged for the docker image maintainers? Sorry, I thought that was this project.

**Reply by michaelklishin** [reply]
The image itself does not control the hostname the container will use.

**Reply by lukebakken** [reply]
> I guess then maybe this should be logged for the docker image maintainer

This is how docker works. You must configure docker to use a consistent hostname if you want a consistent hostname! It is not up to the containers to decide that.

**Reply by gabor-lbl** [reply]
Yes, apologies, I just noticed this described in the "how to use this image" section on dockerhub.

Still, if you feel like adding a warning in the logs about this, one way it could be done is checking if the code is running in docker and if the hostname is the same as the container id. Inside the container /proc/self/mountinfo will contain the container id. So, something like:

```
cat /proc/self/mountinfo  | grep docker | grep `hostname`
```

will return information if the warning should be shown.

Thanks!
